lib/db/db_sqlite.rb

```diff
@@ -0,0 +1,85 @@
+require 'db/db'
+require 'sqlite3'
+
+class DBSQLite < DB
+
+  def initialize(sqlite, table, columns)
+    @sqlite  = sqlite
+    @table   = table
+    @columns = columns
+    create_table
+  end
+
+  def size
+    all.size
+  end
+
+  def all
+    contacts = sqlite.execute "SELECT * FROM " << table
+    contacts.map { |contact| Hash[columns_with_id.zip(contact)] }
+  end
+
+  def at(index)
+    all.fetch(index, {})
+  end
+
+  def add(contact)
+    sql = "INSERT INTO " << table << field_columns << " VALUES(" << field_values(contact) << ")"
+    sqlite.execute sql
+  end
+
+  def update(contact)
+    sql = "UPDATE " << table << " SET " << field_setters(contact) << " WHERE id=" << contact["id"].to_s
+    sqlite.execute sql
+  end
+
+  def search(term)
+    all.select { |contact| contact.any? {|key, val| val.to_s.include?(term)} }
+  end
+
+  def index_of_id(id)
+    all.index { |contact| contact["id"] == id }
+  end
+
+  def close
+    sqlite.close if sqlite
+  end
+
+  private
+
+  attr_reader :sqlite, :table, :columns
+
+  def create_table
+    sql = "CREATE TABLE IF NOT EXISTS " << table << "(id INTEGER PRIMARY KEY, " << fields << ")"
```

Comment:
This is how I had always done it... I thought it was a good idea so that if the schema changes, (like, if I add more columns or remove columns, etc.) I don't have to do a lot of changes.